oriedita/src/main/java/oriedita/editor/canvas/impl/CreasePattern_Worker_Impl.java

```diff
@@ -757,16 +763,38 @@ public int getCandidateSize() {
         return lineCandidate.size();
     }
 
+    @Override
+    public void setIsSelectionEmpty(boolean isSelectionEmpty){
+        boolean oldIsSelectionEmpty = this.isSelectionEmpty;
+        this.isSelectionEmpty = isSelectionEmpty;
+        this.pcs.firePropertyChange("isSelectionEmpty", oldIsSelectionEmpty, isSelectionEmpty);
+    }
+
+    @Override
+    public boolean getIsSelectionEmpty(){ 
+        return isSelectionEmpty;
+    }
+
+    @Override
+    public void refreshIsSelectionEmpty(){
+        boolean latestState = foldLineSet.isSelectionEmpty();
+        if(latestState != this.isSelectionEmpty){
+            setIsSelectionEmpty(latestState);
+        }
+    }
+
     @Override
     public void select_all() {
         foldLineSet.select_all();
+        refreshIsSelectionEmpty();
     }
 
     @Override
     public void unselect_all(boolean ignorePersistent) {
         if (!applicationModel.getSelectPersistent() || ignorePersistent) {
             foldLineSet.unselect_all();
         }
+        setIsSelectionEmpty(true);
```

Comment:
This line needs to be moved into the if statement to prevent a bug. 
`ApplicationModel.selectPersistent` is a setting that allows you to keep the selection when changing to tools that would usually clear it (e.g. drawing tools, set the Persist checkbox in the bottom left to test it out). The tools still call `unselect_all`, but the actual unselecting is only done if the setting is set to false (or if the caller specifically says to ignore the setting).
 
Currently, if you enable the Persist Checkbox, select something and then select another tool, the buttons get disabled even though the selection isn't empty.

Reply:
Aha, good catch, thank you!

I wasn't aware of this button and feature - the existing code makes a lot more sense now.

I'm currently away from my PC for another few days due to Eid holidays. I will patch, test and update this PR as soon as I get back.